aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs

```diff
@@ -377,8 +377,79 @@ public async Task<GridResult<GetRequestEmployeeDto>> GetEmployeesInSalaryRequest
             else if (input.BranchIds != null && input.BranchIds.Count > 1) query = query.Where(x => input.BranchIds.Contains(x.BranchId));
             return await query.GetGridResult(query, input.GridParam);
         }
+public async Task<List<string>> CreateSalaryChangeRequestFromCheckpointTool(CreateSalaryChangeRequestFromCheckpointDto input)
+        {
+            var newChangeRequest = Create(new CreateSalaryRequestDto
+            {
+                Name = input.Name ?? "Checkpoint",
+                ApplyMonth = new DateTime(input.ApplyMonth.Year, input.ApplyMonth.Month, 1)
+            });
+
+            var dictLevel = WorkScope.GetAll<Level>()
+                                      .Select(s => new { s.Code, s.Id })
+                                      .ToList()
+                                      .GroupBy(s => s.Code.ToLower().Trim())
+                                      .ToDictionary(s => s.Key, s => s.FirstOrDefault());
+            
+            var dicEmployee = WorkScope.GetAll<Employee>()
+              .Where(x => x.Status != EmployeeStatus.Quit)
+              .Select(x => new
+              { 
+                  x.Id,
+                  x.Branch,
+                  x.Level,
+                  x.JobPosition,
+                  x.Email,
+                  x.Salary,
+                  x.RealSalary,
+                  x.UserType,
+              })
+              .ToList()
+              .GroupBy(x => x.Email.ToLower().Trim())
+              .ToDictionary(x => x.Key, s => s.FirstOrDefault());
+
+            var listResult = new List<string>();
+            var listRequestChangeSalary = new List<SalaryChangeRequestEmployee>();
+            long employeeId;
+            double realSalary;
+
+            foreach (var dto in input.RequestChangeSalaryEmployees)
+            {
+                try
+                {
+                    if (!dictLevel.ContainsKey(dto.ToLevelCodeToLowerTrim))
```

Comment:
if (!dictLevel.ContainsKey(dto.ToLevelCodeToLowerTrim))
                    {
                        listResult.Add($"{dto.EmailAddress} - {dto.ToLevelCode} - fail: not found level code in HRM");
                        continue;
                    }